functions/java-jre.bash

```diff
@@ -253,23 +272,28 @@ java_zulu_fetch(){
 ##
 ##    java_zulu_update_available(String arch)
 ##
-java_zulu_update_available(){
+java_zulu_update_available() {
   if [ ! -x "$(command -v java)" ] || [[ "$(java -version 2>&1> /dev/null)" == *"AdoptOpenJDK"* ]]; then return 0; fi
+
   local availableVersion
-  local requestedArch
-  local javaVersion
-  local javaArch
   local filter
-  local link8
+  local javaArch
+  local jdkBin
+  local javaVersion
   local link11
-  if [ ! -x "$(command -v jq)" ]; then
-    cond_redirect apt-get install --yes jq
+  local link8
+  local requestedArch
+
+  if ! [ -x "$(command -v jq)" ]; then
+    echo -n "$(timestamp) [openHABian] Installing Java Zulu prerequisites (jq)... "
+    if cond_redirect apt-get install --yes jq; then echo "OK"; else echo "FAILED"; return 1; fi
   fi
 
   filter='[.zulu_version[] | tostring] | join(".")'
-  link8="https://api.azul.com/zulu/download/community/v1.0/bundles/latest/?jdk_version=8&ext=tar.gz&os=linux"
+  jdkBin="$(find /opt/jdk/*/bin ... -print -quit)"
+  javaVersion="$("${jdkBin}"/java -version |& grep -m 1 -o "[0-9]\{0,3\}\.[0-9]\{0,3\}\.[0-9]\{0,3\}\.[0-9]\{0,3\}")"
   link11="https://api.azul.com/zulu/download/community/v1.0/bundles/latest/?jdk_version=11&ext=tar.gz&os=linux"
-  javaVersion=$("$(find /opt/jdk/*/bin ... -print -quit)"/java -version |& grep -m 1 -o "[0-9]\{0,3\}\.[0-9]\{0,3\}\.[0-9]\{0,3\}\.[0-9]\{0,3\}")
+  link8="https://api.azul.com/zulu/download/community/v1.0/bundles/latest/?jdk_version=8&ext=tar.gz&os=linux"
```

Comment:
Code duplication, see link8 and link11 in function before (or I overlooked a subtile difference in both strings). Should we try to avoid that, especially as we learned that this location changes often and needs to be patched at two places?

Reply:
No, that is just the base URL for the API it does not change. I found that using to separate links was the most elegant way to serve the two versions.

Reply:
It's fine to separate 8 and 11. I was referring to defining both constants link8 and link11 in two functions without sharing it. I know you want to keep them local to the function, but still...